### DISCUSSION TITLE
Problems with drawing normals with color map in section 6.1 of book 1

### DISCUSSION THREAD

**Discussion by kogandg** [RayTracing/raytracing.github.io#1220] [Q&A] [answered]
I was followed the previous section of simply intersecting the sphere correctly, but when showing the normals of the sphere with the color map, it seems that they are flipped or swapped in some way.
This is the output:
![out](https://github.com/RayTracing/raytracing.github.io/assets/19920965/a6248e2a-1b89-4536-bece-69b4e53f4a12)
This is the expected output from the book:
![img-1 04-normals-sphere](https://github.com/RayTracing/raytracing.github.io/assets/19920965/5764e0b1-992a-4463-b323-50c6b38ce067)
Does anyone know a possible bug that would cause this problem?
Here is the commit that added the sphere and has this problem: https://github.com/kogandg/Renderer/commit/64472cd8dd6b79d5daeb24d449ad8fb0550da83c
Here is the link to the whole github: https://github.com/kogandg/Renderer

**Comment by kogandg** [comment, marked as answer]
The problem was with the way I was storing my colors. I was storing them in an unsigned char for ease of writing to the file, but it resulted in the loss of all negative values, which are necessary to avoid the harsh lines and darkness that are seen in the image.  The fixed commit is: https://github.com/kogandg/Renderer/commit/79bfe9fca5e0f47045f52cc332595784a1f54f97

**Comment by hollasch** [comment]
Thank you for following up and letting us all know!
